---
layout: default
---

<style>
img {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
h1 {
  text-align: center;
  }
a:link {
  color: deepskyblue;
  background-color: transparent;
  text-decoration: none;
}
a:hover {
  color: deepskyblue ;
  background-color: transparent;
  text-decoration: underline;
}

a.tablelink:link {
  color: #DAF7A6;
  background-color: transparent;
  text-decoration: none;
  }
  a.tablelink:hover {
  color: #DAF7A6;
  background-color: transparent;
  text-decoration: underline;
}
  
.myDiv {
  padding: 10px;
  border-radius: 20px;
  background-color: #75b5aa;
  color: black;
}
.myDiv2 {
  padding: 10px;
  border-radius: 20px;
  background-color: lavender;
  color: black;
  }
 .myDiv3 {
   padding: 10px;
   border-radius: 20px;
   background-color: #aa759f;
   color: black;
  }
hr.custom {
  border-top: 1px dashed ;
  }
  table {
  width:100%;
}
table, th, td {
  
  border: 3px solid black;
  border-collapse: collapse;
  
}
th, td {
  padding: 15px;
  text-align: left;
}

th { 
  background-color: #303030;
  color: #f4bf75;
  }
  
tr {
 background-color: #006150
}  


.navbar {
  width: 100%;
  overflow: hidden;
  background-color: #005858;
  border: 4px solid #b5e853; 
  border-radius:10px
}

.navbar a {
  float: left;
  font-size: 16px;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

.dropdown {
  float: left;
  overflow: hidden;
}

.dropdown .dropbtn {
  font-size: 16px;  
  border: none;
  outline: none;
  color: white;
  padding: 14px 16px;
  background-color: inherit;
  font-family: inherit;
  margin: 0;
}

.navbar a:hover, .dropdown:hover .dropbtn {
  background-color: #f4bf75;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
}

.dropdown-content a {
  float: none;
  color: #005858;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content a:hover {
  background-color: #ddd;
}

.dropdown:hover .dropdown-content {
  display: block;
}
</style>



<p><a href="https://usfomicshub.github.io/"></p>

<p align="center">
<img src="https://github.com/usfomicshub/usfomicshub.github.io/blob/master/img/Hub_logo_2.0_small.PNG?raw=true" width="300" height="250" class="center">
 </p>

<p></a></p>

<h1 align="center">Welcome to the USF Omics Hub git repository!</h1>

  <h3 align="center"><p style = "color: #f4bf75">We post our materials for our data-analysis workshops here, information for upcoming events, as well as the code we've developed for various other omics-related data analyses. </p></h3>

<div class="navbar">

  <div class="dropdown">
    <button class="dropbtn">USF Genomics
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics">USF Genomics Program</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/USF_Genomics_Core/index.html">USF Genomics Core: Facilities & Resources</a>
   </div>
  </div> 

  <div class="dropdown">
    <button class="dropbtn">Hub Service Inquires
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/request-services">Request Services</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Policies_and_Procedures/index.html">Service Policies & Procedures</a>
   </div>
  </div> 


  
  <div class="dropdown">
    <button class="dropbtn">Hub Workshops
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/genomics-training">Training Workshops Overview</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/index.html">Microbiome Workshop (June 2020)</a>
   </div>
  </div> 


<a href="https://usfomicshub.github.io/usfomicshub.github.io/Cluster_Computing/index.html">Cluster Computing</a> 


 
<a href="https://usfomicshub.github.io/faq">FAQ</a> 


</div>


<p></a></p>
                                                                                                                                     
<h1 id="microbiome-data-analysis-workshop">Microbiome Data-Analysis Workshop</h1>
<h2 id="goal"><span style = "color: white"> GOAL</span></h2>
<blockquote>
<p>In addition to understanding the importance of experimental design, we will walk through turning raw sequence data into useful counts-data that we can use to visualize microbiome sample-composition. We will then discuss methods to extrapolate function from these abundance-data (and limitations thereof) to ultimately arrive at biological insight.</p>
</blockquote>


<h3 id="table-of-contents">Table of Contents</h3>
<ol>
  <a href = "#pre-course-materials"><li>Pre-course Materials</a>
  <a href = "#day-one"><li>Day One<ul></a>
  <a href = "#presentation-slides"><li> Presentation Slides</li></a>
</ul>
</li>
<a href = "#day-two"><li>Day Two<ul></a>
  <a href = "#day-two"><li>Presentation Slides</li></a>
  <a href = "#generating-asv-tables-from-microbiome-sample-sequencing-data"><li>Generating ASV Tables from Microbiome-sample Sequencing Data</li></a>
</ul>
</li>
<a href="#day-three"><li>Day Three<ul></a>
  <a href="#day-three"><li>Presentation Slides</li></a>
  <a href="#microbiome-data-visualization"> <li> Microbiome Data Visualization </li></a>
</ul>
</li>
<a href = "#resources"><li>Resources</a>
</li>
</ol>

<h2 id="pre-course-materials">Pre-course Materials</h2>
<hr class = "custom">

<p> <span style = "color: #b5e853">&#11043;</span> <a href="https://www.nature.com/articles/s41579-018-0029-9.pdf">  Best Practices for Analyzing Microbiomes</a> - This article discusses how all stages of conducting a microbiome study, from designing the experiment to collecting and storing the samples to obtaining
insight from graphical displays of the sequence data, can substantially impact the result.
<p> <span style = "color: #b5e853"> &#11043; </span> If you do not have R and RStudio installed already, you can follow these<a href="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/precoursematerials/r_installation_instructions_062020.pdf" download = "RInstallationInstructions"> instructions</a> for both Mac and Windows. If you already have RStudio, make sure you're using R version <span style = "color: #aa759f">4.0.1 -- "See Things Now"</span> by clicking on <span style = "color: black; background-color: lightgrey; padding: 2px;font-family: Helvetica;">Global Options...</span> in the <span style = "color: black; background-color: lightgrey; padding: 2px;font-family: Helvetica;">Tools</span> tab. The version is also stated in the first line of the console when you first open RStudio. </p>
<p> You can copy the library with your old packages from the previous version to the new version then update the packages by clicking <span style = "color: black; background-color: lightgrey;padding:2px;font-family: Helvetica;">Check for Package Updates...</span> also in the <span style = "color: black; background-color: lightgrey;padding=2px;font-family: Helvetica;">Tools</span> tab within RStudio.</p>
<p> If you have Windows, then a very easy way to update your R-version and packages is by simply running the following code in the RStudio console: </p>
<pre><code>install.packages(<span class="hljs-string">"installr"</span>)

<span class="hljs-function"><span class="hljs-title">library</span><span class="hljs-params">(installr)</span></span>

<span class="hljs-function"><span class="hljs-title">updateR</span><span class="hljs-params">()</span></span>
</code></pre>
<p> <span style = "color: #b5e853"> &#11043; </span> We'll be moving quickly through basic concepts in R to get to the actual data-analysis. We strongly recommend reviewing these two R tutorials to get you started/help you keep up: 
  <br>
  <ul>
  <a href= "https://www.datacamp.com/community/tutorials/basic-programming-skills-r"><li>Basic Programming Skills in R</a></li> 
  <br>
<a href = "https://www.datacamp.com/community/tutorials/utilities-in-r"><li>Utilities in R </li></a>
</ul>
<br>
<br>
  
<h2 id="day-one">Day One</h2>
<hr class = "custom">
<h3 id="presentation-slides">Presentation Slides</h3>
Day one of this workshop focuses on an overview of available methods and best-practice considerations for experimental design in microbiome studies. You can download the individual presentations for each topic in the agenda below or you can view them <a href = "https://github.com/usfomicshub/microbiome_workshop/tree/master/slides/day1">here.</a></p>
<p>A more detailed schedule of the workshop should have been emailed to you. You can also download it <a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/Microbiome_Analysis_Training_Agenda_June_2020.xlsx download = "workshopschedule"> here</a>.
<table>
  <tr style="text-align:center">
    <th>AGENDA</th>
    <th>INSTRUCTOR</th> 
  </tr>
  
  <tr>
    <td><a href= "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/JO.Hub.Microbiome.06102020.pdf" class="a.tablelink" download = "USFGenomicsInto">USF Genomics Introduction and Workshop Overview</a></td>
    <td><span style = "color: white">Dr. Jenna Oberstaller</span></td>
  </tr>
  <tr>
  <td><a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/USF%20Genomics%20Core%20Introduction-06102020-MZ.pdf" class="a.tablelink" download = "USFGenomicsEquipCore"> USF Genomics Equipment Core</a></td>
<td><span style = "color: white">Dr. Min Zhang</span></td>  
  </tr>
  <tr>
    <td><a href="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/Microbiome_talk_AS_Jun10_2020.pdf" class="a.tablelink" download = "IntroMDA"> Introduction to Microbiome Data Analysis</a></td>
    <td><span style = "color: white">Dr. Anujit Sarkar</span></td>
  </tr>
    <tr>
    <td><a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/Best%20practices%20for%20microbiome%20sample-handling%20and%20nucleic%20acid-processing.pdf" class="a.tablelink" download = "BestPracticesMBsampling">Best Practices for Microbiome Sample-handling and Nucleic Acid-processing </a></td>
    <td style = "color: white">Swamy Rakesh Adapa, MS</td>
  </tr>
  <tr>
  <td><a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/Statistical%20considerations%20of%20microbiome%20studies.pdf" class="a.tablelink" download = "StatisticalConsiderations">Statistical Considerations for Microbiome Studies</a></td>
  <td style = "color: white">Dr. Ming Ji</td>
  <tr>
    <td><a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/Microbiome_ED_compressed.pdf" class="a.tablelink"> Experimental Design</a></td>
    <td style = "color: white">Swamy Rakesh Adapa, MS</td>
  </tr>
  <tr>
  <td><a href ="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/jg_microbiome_plots_march2020.pdf" class="a.tablelink"> Overview of Microbiome Data-Visualization  </a></td>
  <td span style = "color: white">Dr. Justin Gibbons</span></td>
  </tr>
    <tr>
  <td><a href ="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day1/Functional_profiling_with_Picrust2_workshopTEK_June2020.pdf" class="a.tablelink">Functional Profiling with PICRUSt2</a></td>
  <td span style = "color: white">Dr. Thomas Keller</span></td>
  </tr>
  </table>
  <br>
  <br>


<h2 id="day-two">Day Two</h2>
<hr class = "custom">
<h3 id="presentation-slides">Presentation Slides</h3>
<p><a href ="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day2/R%20Tutorial.pdf">Introduction to R and Plotting Data (Dr. Charley Wang)</a></p>
<p><a href = "https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day2/AS_11June_talk_dada2.pdf"> Taxonomic Analysis (Dr. Anujit Sarkar)</a></p>

<h3 id="generating-asv-tables-from-microbiome-sample-sequencing-data">Generating ASV Tables from Microbiome-sample Sequencing Data</h3>
<h3 id="pipeline-overview"><span style = "color: #d28445">PIPELINE OVERVIEW</span></h3>
<div class = "myDiv">
<p><strong>Goal:</strong> The purpose of this analysis is to obtain an Amplicon Sequence Variant (ASV) table for all of our microbiome-sample example-data.</p>
<p><strong>Input data:</strong> We will start with demultiplexed fastq files for all samples. This analysis is for paired-end data. Thus, for each sample, there will be two files, named according to Illumina platform conventions:</p>
<ul>
<li>Forward-reads, named *_R1_001.fastq </li>
<li>Reverse-reads, named *_R2_001.fastq</li>
   </ul>
</div>
<p>
</p>
  <h3 id="creating-the-project"><span style = "color: #d28445">CREATING THE PROJECT</span></h3>
  <div class = "myDiv3">
  <p><strong>1.</strong> Follow this <a href="http://usf.box.com/s/brtr5qpq3pg63ngs85yok162q6ovkjw7">link </a> and click download in the upper right corner.</p>
  <p><strong>2.</strong> Extract the downloaded zip file to where you want it. </p>
  <p><strong>3.</strong> Open RStudio and click on <span style = "color: black; background-color: lightgrey;padding:2px;font-family: Helvetica;">New Project</span> in the <span style = "color: black; background-color: lightgrey;padding:2px;font-family: Helvetica;">File tab</span>.</p>
  <p><strong>4.</strong> Create the new project by choosing 'Existing Directory'</p>
  <p><strong>5.</strong> Browse to the directory where you extracted the zip file and make sure 'day2' is the last folder in your project directory file path.</p>
  <p> You should see the folders(Ranalysis,Rdata,etc..) when you open the day2 folder.</p>
  <p><strong>6.</strong> Click create project.</p>
  </div class = "myDiv3">
 <p>
  </p>
  <br>
  <p> For more info behind the logic of creating RStudio projects and adhering to an organizational directory-structure as you build your data-analysis skills, see this <a href="https://swcarpentry.github.io/r-novice-gapminder/02-project-intro/">post on reproducible scientific data-analyses</a> from Software Carpentry.
  We don't use exactly the same structure they do, but the concepts are the same: structured analyses make sharing and reproducing analyses much easier!
  </p>
  <br>
<h3 id="tutorial-structure"><span style = "color: #d28445">TUTORIAL STRUCTURE</span></h3>

<p>Before we begin, let’s take a moment to get organized. The importance of documentation and good record-keeping are essential to producing high-quality and reproducible computational analyses, just as they are at the bench!</p>

<p>We recommend you keep your analyses organized by project (just as we organized this example).
<p>Looking around in the file browser tab of the lower right section, you should find the following folders if you set the project directory correctly:</p>

<div class = "myDiv2">
<p><strong>Rdata</strong>: this folder contains our input .fastq.gz files and our input database of 16S-sequences that we’ll use to identify taxa present in our samples.</p>
<p><strong>Ranalysis</strong>: this folder contains any scripts we create to analyze our data, like this R-Markdown (.Rmd) document.</p>
<p><strong>Routput</strong>: we will direct any output data-files from our analyses to this folder.</p>
<p><strong>Rfigs</strong>: we will direct any figures we generate from our analyses to this folder.</p>
<p><strong>Rsource</strong>: this folder contains any R source-scripts we create to set up our environment for our analyses–custom functions, which packages to load, etc. etc. You don’t need to worry about this one since we made it for you.</p>
<p>You can think of any files in Rsource as set-up scripts--just load it at the beginning of your session and forget about it.</p>
</div class = "myDiv2">  
<p>
</p>
<h3 id="setting-up-the-environment"><span style = "color: #d28445">SETTING UP THE ENVIRONMENT</span></h3>
<p> Now that we are familar with the project, we can set up the environment!<p>
  <p style = "text-indent:20px"><span style = "color: #aa759f"> 1. Go to the <strong>Ranalysis</strong> folder in the file browser in the lower right section and open the .Rmd file</span></p>
  <p <p style = "text-indent:20px"><span style = "color: #aa759f"> 2. Run only the first chunk of code beginning at line 10 by clicking the green arrow within the upper right corner of the chunk.</span></p>
 <p> Running this code calls a source script from the RSource folder that installs all of the packages needed to run the tutorial. </p>
  
  <p> As you run the tutorial chunk by chunk, you can follow along with the output document linked below that includes the R code and its output.</p>
  
  <h3><a href = "https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/microbiome_workshop_demos/day2/Ranalysis/#begin-analyses">Let's begin the day2 tutorial!</a></h3>
<br>
  <br>



<h2 id="day-three">Day Three</h2>
<hr class = "custom">
<h3 id="presentation-slides">Presentation Slides</h3>
<p><a href="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day3/jg_microbiome_plots_march2020.pdf">Taxonomic Analysis: Visualization Part I (Dr. Justin Gibbons) </a></p> 
<p><a href="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day3/Functional_profiling_with_Picrust2_workshopTEK_June2020%20.pdf">Taxonomic Analysis: Visualization Part II (Dr. Thomas Keller) </a></p>
<p><a href="https://github.com/usfomicshub/microbiome_workshop/raw/master/slides/day3/Machine%20learning%20for%20the%20microbiome_tek_jun2020.pdf">Taxonomic Analysis: Machine Learning (Dr. Thomas Keller)  </a></p>
<h3 id="microbiome-data-visualization">Microbiome Data Visualization</h3>
<h3 id="pipeline-overview"><span style = "color: #d28445">PIPELINE OVERVIEW</span></h3>
<div class = "myDiv">
<p><strong>Goal:</strong> First, we will plot our results from the small data-set we analyzed yesterday. Then we will visualize other microbiome data (previously processed in the same way as we processed the small test-dataset from yesterday, from raw sequence-data to OTU or ASV tables) from a different larger dataset.</p>
  <p><strong>Input data:</strong> </p>
  <p style = "color:#8f5536"><strong>Taxonomic Analysis: Visualization Part I</strong><p>
  <p>Our outputs from Day 2 (we've provided them in the Day 3 download)</p>
<ul>
<li>demo_asv_counts.tsv</li>
  <li>demo_asvs_taxonomy.tsv</li>
  <li>made_up_sample_data.tsv</li>
</ul>
  <p>Biom file and mapping files that will be converted to phyloseq-class</p>
 <ul>
  <li>otu_table.biom</li>
  <li>meta_data.csv</li>
  </ul>
  <p style = "color:#8f5536"><strong>Taxonomic Analysis: Visualization Part II</strong><p>
  <ul>
<li>demo_asv_counts.tsv</li>
  <li>pathways_out -> pathway .tsv files</li>
  <li>metagenome_out -> kegg .tsv files</li>
  <li>metadata.tsv</li>
  </ul>
  <p style = "color:#8f5536"><strong>Taxonomic Analysis: Machine Learning</strong><p>
  <ul>
  <li>an OTU table (converted to relative abundance)</li>
  <li>a table of metadata to associate with the OTUs</li>
</ul>
</div>
  
 <p>
</p>
  
  <h3 id="creating-the-project"><span style = "color: #d28445">CREATING THE PROJECT</span></h3>
  <div class = "myDiv3">
  <p> <strong> We'll follow the same steps to create the day3 project in RStudio as we did to create the day2 project. </strong></p>
  <p><strong>1.</strong> Follow this <a href="https://usf.box.com/s/eeze6tggv4ciyxjzplygaew4wzke3aa3">link</a> and click download in the upper right corner.</p>
  <p><strong>2.</strong> Extract the downloaded zip file to where you want it. </p>
  <p><strong>3.</strong> Open RStudio and click on <span style = "color: black; background-color: lightgrey;padding:2px;font-family: Helvetica;">New Project</span> in the <span style = "color: black; background-color: lightgrey;padding:2px;font-family: Helvetica;">File tab</span>.</p>
  <p><strong>4.</strong> Create the new project by choosing 'Existing Directory'</p>
  <p><strong>5.</strong> Browse to the directory where you extracted the zip file and make sure 'day3' is the last folder in your project directory file path.</p>
  <p> You should see the folders(Ranalysis,Rdata,etc..) when you open the day3 folder.</p>
  <p><strong>6.</strong> Click create project.</p>
  </div class = "myDiv3">
 <p>
  </p>
  <h3 id="tutorial-structure"><span style = "color: #d28445">TUTORIAL STRUCTURE</span></h3>

  <p> Day 3 follows the same directory-structure as Day 2 above. </p>

</p>
  <h3 id="setting-up-the-environment"><span style = "color: #d28445">SETTING UP THE ENVIRONMENT</span></h3>
<p> If you succesfully ran the Day 2 tutorial, then you should already have the packages needed for Day 3. </p>
  <p> However, the same instructions for Day 2 go for Day 3 if needed.</p> 
  <p>As you run the tutorial chunk by chunk, you can follow along with the output document linked below that includes the R code and its output.</p>
  <p><h3><a href = "https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/microbiome_workshop_demos/day3/Ranalysis/PartI/#microbiome-workshop-data-analysis-and-visualization">Let's begin the day3: visualization part I tutorial!</a></h3></p>
    <h3><a href = "https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/microbiome_workshop_demos/day3/Ranalysis/PartII/#usf-omics-hub-microbiome-workshop-day-3-part-ii-functional-analyses">Let's begin the day3: visualization part II tutorial!</a></h3>
<h3><a href = "https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/microbiome_workshop_demos/day3/Ranalysis/PartIII/#usf-omics-hub-microbiome-workshop-day-3-part-ii-machine-learning-approaches-to-functional-analyses">Let's begin the day3: machine learning tutorial!</a></h3>
<br>
  <br>
  
  
<h2 id="resources">Resources</h2>
<hr class = "custom">
 <p>
 <p style = "color: #aa759f">Here are some resources mentioned in this workshop and some extra information that you might find helpful in your microbiome research. </p>
  <p><details closed></p>
<p><summary>Journal-articles Dr. Ji referenced as examples for different study-designs in his session (Statistical Considerations for Microbiome Studies)</summary></p>
<ul>
  </p><li><a href="https://msystems.asm.org/content/msys/4/4/e00325-19.full.pdf">A Cross-Sectional Study of Compositional and Functional Profiles of Gut Microbiota in Sardinian Centenarians</a></li></p>

<p><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7040164/pdf/fnins-14-00127.pdf">Case-Control Study of the Effects of Gut Microbiota Composition on Neurotransmitter Metabolic Pathways in Children With Attention Deficit Hyperactivity Disorder</a></li></p>

<p><li><a href="https://www.mdpi.com/2076-2607/7/1/17/html">Socioeconomic Status and the Gut Microbiome: A TwinsUK Cohort Study</a></li></p>

<p><li><a href="https://www.nature.com/articles/s41366-019-0440-y?proof=true">Structured exercise alters the gut microbiota in humans with overweight and obesity—A randomized controlled trial</a></li></p>

<p><li><a href="https://www.nature.com/articles/s41467-017-01973-8">Meta-analysis of gut microbiome studies identifies disease-specific and shared responses</a></li></p>
</ul>
</details>
  <details closed>
  </p><summary>Microbiome R Packages</summary></p>
<ul>
<p><li><a href="https://www.bioconductor.org/packages/release/bioc/manuals/dada2/man/dada2.pdf">Dada2 Reference Manual</a></li></p>
  <p><li><a href="https://vaulot.github.io/tutorials/Phyloseq_tutorial.html">PhyloSeq Tutorial</a></li></p>
<p><li><a href="https://bioconductor.org/packages/release/bioc/vignettes/SIAMCAT/inst/doc/SIAMCAT_vignette.html">SIAMCAT vignette</a></li></p>
</ul>
 </details closed>
  <details closed>
  <p><summary>Microbiome Software</summary></p>
  <ul>
  <p><li><a href ="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>: 	A quality control tool for high throughput sequence data.</li></p>
  <p><li><a href = "https://qiime2.org/">QIIME2</a>: an open-source bioinformatics pipeline for performing microbiome analysis from raw DNA sequencing data.</li></p>
  <p><li><a href="https://github.com/picrust/picrust2/wiki/PICRUSt2-Tutorial-(v2.3.0-beta">PICRUSt2 Tutorial</a></li></p>
  </ul>
  </details>
  </p>
  
 <br>
  <br>
  
  
      <h2 id="certificate-of-completion">Certificate Of Completion</h2> 
  <hr class = "custom">

  <details closed>  
  <p><summary>Please open this tab and take the survey to receive your <span style = "color: #d28445">USF Omicshub Microbiome Data Analysis Certificate</span>!</summary></p>
  <iframe src="https://docs.google.com/forms/d/e/1FAIpQLSdiBLf1qAT0FeqYW-At-vYziyQfakAeR7AvEippadrH5co8KQ/viewform?embedded=true" width="640" height="2990" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
  </details>


