<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="/assets/css/style.css?v=3b1ea14f0e4b9602ac90929571ec4593591de56a">

<!-- Begin Jekyll SEO tag v2.6.1 -->
<title>Saving_loading_if_for_apply | USF Omics Hub</title>
<meta name="generator" content="Jekyll v3.8.5" />
<meta property="og:title" content="Saving_loading_if_for_apply" />
<meta name="author" content="Chengqi(Charley) Wang" />
<meta property="og:locale" content="en_US" />
<meta name="description" content="USF Omics Hub official github-repository: your landing-spot for finding our USF Genomics community-resources." />
<meta property="og:description" content="USF Omics Hub official github-repository: your landing-spot for finding our USF Genomics community-resources." />
<link rel="canonical" href="http://localhost:4000/R_training/Saving_loading_if_for_apply.Rmd" />
<meta property="og:url" content="http://localhost:4000/R_training/Saving_loading_if_for_apply.Rmd" />
<meta property="og:site_name" content="USF Omics Hub" />
<script type="application/ld+json">
{"author":{"@type":"Person","name":"Chengqi(Charley) Wang"},"headline":"Saving_loading_if_for_apply","description":"USF Omics Hub official github-repository: your landing-spot for finding our USF Genomics community-resources.","@type":"WebPage","url":"http://localhost:4000/R_training/Saving_loading_if_for_apply.Rmd","@context":"https://schema.org"}</script>
<!-- End Jekyll SEO tag -->

  </head>

  <body>

    <header>
      <div class="container">
        <h1>usfomicshub.github.io</h1>
        <h2>USF Omics Hub official github-repository: your landing-spot for finding our USF Genomics community-resources.</h2>

        <section id="downloads">
          
          <a href="https://github.com/usfomicshub/usfomicshub.github.io" class="btn btn-github"><span class="icon"></span>View on GitHub</a>
        </section>
      </div>
    </header>

    <div class="container">
      <section id="main_content">
        <br/><br/>

*date: 1/29/2020*
####Load the RNA-Seq data
```{r}
##set your working directory and load the data
setwd('~/Documents/work/paper_write/genome_training/R_training/')
df <- read.table('vehicle_drug_feature_counts.txt',
                 header = T, sep = '\t', row.names = 1)
summary(df)
df <- df[,6:9]
```
<br/><br/>

####Save the data
```{r}
write.table(df, 
            file = 'vehicle_drug_feature_counts.copy.txt.txt', #either a character string naming a file or a connection open for writing. "" indicates output to the console.
            row.names = T, #indicating whether the row names of df are to be written
            col.names = T, #indicating whether the column names of df are to be written
            sep = '\t',    #the field separator string
            quote = F      #a logical value (TRUE or FALSE) or a numeric vector. If TRUE, any character or factor columns will be surrounded by double quotes.
            )

###load again
df2 <- read.table('vehicle_drug_feature_counts.copy.txt.txt',header = T, sep = '\t', row.names = 1)
head(df2)
```
<br/><br/>

#### 'For' loop
```{r}
df <- as.matrix(df)
##check the quantile for each assay
for (i in 1:ncol(df))
{
   print(  colnames(df)[i] )
   qu <- quantile( df[, i] )
   print(qu)
}
```
<br/>

#### 'Apply' replaces 'for' loop
```{r}
qu <- apply(df, 2, function(z){quantile(z)})
print(qu)
```
<br/>

#### if...else statement
```{r}
### pick the expressed genes in first assay
exp_cutoff   <- quantile(df[,1], 0.35)
unEXP_cutoff <- quantile(df[,1], 0.3)
n_remaining  <- 0
exp_gene   <- c() #set a empty vector for expressed genes
unexp_gene <- c() #set a empty vector for un_expressed genes
for (i in (1:length(df[,1])))
{
    if (df[i, 1] > exp_cutoff  )
    {
        exp_gene <- c(exp_gene, rownames(df)[i])  
    }
    else if (df[i, 1] < unEXP_cutoff) 
    {
        unexp_gene <- c(unexp_gene, rownames(df)[i])
    }
    else
    {
        n_remaining <- n_remaining + 1    
    }
}

print(c(length(unexp_gene),
      length(exp_gene),
      n_remaining))
```

<br/>

####use 'which' to replace former script
```{r}
exp_gene2 <- rownames(df)[which(df[,1] > exp_cutoff)]
length(exp_gene2)
length(exp_gene)
```



      </section>
    </div>

    
  </body>
</html>
