---
layout: default
---
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<style>
img {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
h1 {
  text-align: center;
  }
a:link {
  color: #63c0f5;
  background-color: transparent;
  text-decoration: none;
}
a:hover {
  color: #63c0f5 ;
  background-color: transparent;
  text-decoration: underline;
}

a.tablelink:link {
  color: #63c0f5;
  background-color: transparent;
  text-decoration: none;
  }
  a.tablelink:hover {
  color: #63c0f5;
  background-color: transparent;
  text-decoration: underline;
}
  
.myDiv {
  padding: 10px;
  border-radius: 20px;
  background-color: #75b5aa;
  color: black;
}
.myDiv2 {
  padding: 10px;
  border-radius: 20px;
  background-color: lavender;
  color: black;
  }
 .myDiv3 {
   padding: 10px;
   border-radius: 20px;
   background-color: #aa759f;
   color: black;
  }
.myDiv4 {
   padding: 10px;
   border-radius: 20px;
   background-color: aliceblue;
   color: #444444;
  }


hr.custom {
  border-top: 1px dashed ;
  }
  table {
  width:100%;
}
table, th, td {
  
  border: 3px solid #303030;
  border-collapse: collapse;
  background-color: black
}
th, td {
  padding: 15px;
  text-align: left;
}

th { 
  background-color: black;
  color: #f4bf75;
  }
  
tr {
 background-color: black;
}  


.navbar {
  width: 100%;
  overflow: hidden;
  background-color: #005858;
  border: 4px solid #b5e853; 
  border-radius:10px
}

.navbar a {
  float: left;
  font-size: 16px;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

.dropdown {
  float: left;
  overflow: hidden;
}

.dropdown .dropbtn {
  font-size: 16px;  
  border: none;
  outline: none;
  color: white;
  padding: 14px 16px;
  background-color: inherit;
  font-family: inherit;
  margin: 0;
}

.navbar a:hover, .dropdown:hover .dropbtn {
  background-color: #f4bf75;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
}

.dropdown-content a {
  float: none;
  color: #005858;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content a:hover {
  background-color: #ddd;
}

.dropdown:hover .dropdown-content {
  display: block;
}
</style>



<a href="https://usfomicshub.github.io/">
<img src="https://github.com/usfomicshub/usfomicshub.github.io/blob/master/img/Hub_logo_2.0_small.PNG?raw=true" width="300" height="250" class="center">
</a>

<p></a></p>

<h1 align="center">Welcome to the USF Omics Hub git repository!</h1>

  <h3 align="center"><p style = "color: #f4bf75">We post our materials for our data-analysis workshops here, information for upcoming events, as well as the code we've developed for various other omics-related data analyses. </p></h3>

<div class="navbar">

  <div class="dropdown">
    <button class="dropbtn">USF Genomics
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics">USF Genomics Program</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/USF_Genomics_Core/index.html">USF Genomics Core: Facilities & Resources</a>
   </div>
  </div> 

  <div class="dropdown">
    <button class="dropbtn">Hub Service Inquires
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/request-services">Request Services</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Policies_and_Procedures/index.html">Service Policies & Procedures</a>
   </div>
  </div> 


  
  <div class="dropdown">
    <button class="dropbtn">Hub Workshops
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/genomics-training">Training Workshops Overview</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/index.html">Microbiome Workshop (June 2020)</a>
   </div>
  </div> 


<a href="https://usfomicshub.github.io/usfomicshub.github.io/Cluster_Computing/index.html">Cluster Computing</a> 


 
<a href="https://usfomicshub.github.io/usfomicshub.github.io/FAQ/index.html">FAQ</a> 


</div>


<p></a></p>


<a href="https://usfomicshub.github.io/">
<img1 src="https://github.com/usfomicshub/usfomicshub.github.io/blob/master/img/Hub_logo_2.0_small.PNG?raw=true" width="300" height="250" class="center">
</a>
                                                                                                                                      
<h1 id="microbiome-data-analysis-workshop">RNA-Seq Data Analysis Computational Workshop</h1>
<h2 id="goal"><span style = "color: white"> GOAL</span></h2>
<blockquote>
<p>This 4-day in-depth course will instruct participants on the theoretical and practical concepts related to RNA sequencing analysis, enabling them to perform these analyses independently.</p>
</blockquote>


<h3 id="table-of-contents">Table of Contents</h3>
<ol>
  <a href = "#pre-course-materials"><li>Pre course Materials</a>
  <a href = "#day-one"><li>Day One<ul></a>
  <a href = "#day-one"><li>Presentation Slides</li></a>
  <a href = "#linux-hands-on-practice"><li>Linux Hands on Practice</li></a>
</ul>
</li>
<a href = "#day-two"><li>Day Two<ul></a>
  <a href = "#day-two"><li>Presentation Slides</li></a>
  <a href = "#read-mapping-hands-on-practice"><li>Read-mapping Hands-on Practice</li></a>
</ul>
</li>
<a href="#day-three"><li>Day Three<ul></a>
  <a href="#day-three"><li>Presentation Slides</li></a>
  <a href="#r-and-bioconductor-hands-on-practice"> <li>R and Bioconductor Hands-on Practice</li></a>
</ul>
</li>

<a href="#day-four"><li>Day Four<ul></a>
  <a href="#day-four"><li>Presentation Slides</li></a>
  <a href="#rna-seq-tuxedo-pipeline-"> <li>RNA-Seq: TUXEDO Pipeline </li></a>
  <a href="#ilab-review"> <li>iLab Review</li></a>


</ol>

<h2 id="pre-course-materials">Pre course Materials</h2>
<hr class = "custom">

<p style = "color:  #FDF5E6"> You should have received an email regarding the agenda and pre-course materials for this workshop as well as been added to the Canvas Course. This canvas course includes all of the installation and tutorials that must be completed prior to the workshop. Those enrolled in this course should also have had a Student Cluster account created. </p>
<p></p>
<p style = "color:  #FDF5E6">These additional resources may also be helpful..</p>
<p></p>
<p style = "color:  #DCDCDC"> <span style = "color: #b5e853"> &#11043; </span>  <a href= "https://wiki.rc.usf.edu/index.php/Connecting_To_SC"> USF Research Computing</a>  offers additional information on connecting to thr SC for MAC and windows. RC also provides a collection Linux tutorials.</p>

<p style = "color:  #DCDCDC"><span style = "color: #b5e853"> &#11043; </span> <a href="https://askubuntu.com/questions/506510/what-is-the-difference-between-terminal-console-shell-and-command-line#:~:text=Shell%20is%20a%20program%20which,software%20%2C%20like%20Gnome%2DTerminal%20.">What is the difference between Terminal, Console, Shell, and Command Line?</a> We hear these terms interchangably but what is the difference? Terminal, shell and command line are often used interchangeably to indicate a text based system for navigating your operating system. Command line is very windows centric terminology, terminal is very mac centric. There are different flavors you can use in a shell including bash. Bash is both a shell and language you can use to interact with the operating system and what we will be using in this course. </p>

</p>

<p> <span style = "color: #b5e853"> &#11043; </span>  <a href=  "https://filezilla-project.org/download.php">FileZilla</a> - FileZilla makes it easy to copy and download files between your local computer and remote clusters.

<p> <span style = "color: #b5e853"> &#11043; </span> If you do not have R and RStudio installed already, you can follow these<a href="https://github.com/usfomicshub/usfomicshub.github.io/raw/master/RNA-Seq_workshop/precoursematerials/r_installation_instructions_062020.pdf" download = "RInstallationInstructions"> instructions</a> for both Mac and Windows.

<p> <span style = "color: #b5e853"> &#11043; </span> We recommend these two R tutorials to get you started: <a href= "https://www.datacamp.com/community/tutorials/basic-programming-skills-r">Basic Programming Skills in R</a> and <a href = "https://www.datacamp.com/community/tutorials/utilities-in-r">Utilities in R </a>


  
<h2 id="day-one">Day One</h2>
<hr class = "custom">

<h3 id ="">Presentation Slides</h3>

<table>
  <tr>
    <th>AGENDA</th>
    <th>INSTRUCTOR</th> 
    
  </tr>
  <tr>
    <td><a href= "" download = "">Intro to Unix</a></td>
    <td><span style = "color: white">Dr. Justin Gibbons</span></td>
  </tr>
  
  <tr>
  <td><a href = "" download = "">RNA-Seq Workflow</a></td>
<td><span style = "color: white">Dr. Charley Wang</span></td>  
  </tr>


  </table>

<h3 id="#linux-hands-on-practice">Linux Hands-on Practice</h3>

Download Charley's <a href = "https://github.com/usfomicshub/RNASeq_workshop_Sept2020/raw/master/slides/day1/LinuxServer.pptx" download = "LinuxServer"> Linux Shell Tutorial</a> to get started.

  <p><strong>Additional Resources:</strong> </p>
  <p> <span style = "color: #b5e853"> &#11043; </span> <a href ="http://windowsbulletin.com/linuxcommands/">Linux Cheat Sheet</a></p>

  <p> <span style = "color: #b5e853"> &#11043; </span> <a href ="http://windowsbulletin.com/vimcheat/">Vim Cheat Sheet</a></p>

<h3 id="pipeline-overview"><span style = "color: #d28445">COPYING TRAINING FOLDER</span></h3>

To prepare for running tomorrow's scripts, we copy a shared folder into our home directory. 

<p></p>
<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day1_jg.png?raw=TRUE" width = 900 style= "border : 5px solid #75b5aa"></img>
<p></p>

<div class = "myDiv4">

In the first line of code shown above by Justin, the cd command by itself takes you to your home directory(any path specified after the cd command will take you to that directory.) You can check this by running "pwd" which prints the directory you are currently in. After making sure you are in your home directory, use "cp -r"
to copy all the contents in the "/shares/biocomputing/Genomics_Training/" folder to your home directory denoted by the period. We can use "ls -lht" to print all the folders and files along with its readable file size sorted by time and date. You should now see "Genomics_Training" in your directory.

</div>
<p></p>

<h2 id="day-two">Day Two</h2>
<hr class = "custom">

<h3 id="day-two">Presentation Slides</h3>

<table>
  <tr>
    <th>AGENDA</th>
    <th>INSTRUCTOR</th> 
    
  </tr>
  <tr>
    <td><a href= "" download = "">Introduction to RNA-Sequencing</a></td>
    <td><span style = "color: white">Dr. Justin Gibbons</span></td>
  </tr>
  
  <tr>
  <td><a href = "" download = "">RNA-Seq Allignment and Visualization</a></td>
<td><span style = "color: white">Dr. Charley Wang</span></td>  
  </tr>

  </table>

<h3 id ="read-mapping-hands-on-practice">Read-mapping Hands-on Practice</h3>
 
  Now, we run the scripts within the "Gemonics_Training" directory that we copied yesterday.To get to the scripts, we can run the following lines from our home directory.

<p></p>

<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt1.png?raw=TRUE" width=900 style= "border : 5px solid #75b5aa">

<p></p>




The My_Scripts folder contains six shell-scripts comprising the UNIX portions of the RNAseq data-analysis pipeline we run during the workshop. The scripts should be edited appropriately with the email-settings edited to YOUR email address.
<p></p>
The scripts are numbered in the order you should run them (via commandline, while logged-in to your student-cluster account). The programs they call and purpose of each step:
<p></p>
<ul>
<li><strong>01_check_fastq_qc.sh</strong>: quality control checks on raw sequence</li>
<li><strong>02_build_hisat_index.sh</strong>: Build an index of your reference-genome</li>
<li><strong>03_run_hisat.sh</strong>: Align reads to your indexed reference-genome</li>
<li><strong>04_run_cufflinks.sh</strong>: Assemble reads into transcripts</li>
<li><strong>05_run_cuffnorm.sh</strong>: Get normalized gene counts</li>
<li><strong>06_run_featureCounts.sh</strong>: Get raw gene counts</li>
</ul>
<p></p>
<div class = "myDiv2" style="border: 5px solid #303030">
<strong>Lets Begin!</strong>
<p></p>

Run "vim" to open a file in Vim editor. To insert and modify text, you can enter INSERT mode by pressing the i key. To exit and return to normal mode, hit the escape key. To save changes and quit, press the colon in normal mode to switch to Command Line mode then type "wq" or w to just save(:w) or q to just quit without making changes(:q). <i>*Do not forget to change the mail-user settings to your usf email address!</i>
<p></p>
<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke"> vim 01_check_fastq_qc.sh
</code></pre>
<p></p>
After saving and quiting, we can submit this job using "sbatch" as shown below. If you specified your email, then you should receive an email confirming the submitted job and whether it was successful. If the job was successful, then you should see the output files in the output directory, "./Genomics_Training/FASTQ_QC". 

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke"> sbatch 01_check_fastq_qc.sh
</code></pre>
<p></p>

Since there are a lot of files, we can combine the files using MultiQC but we will need to install miniconda first. 

<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke"> echo ". /apps/miniconda/3.6.1/etc/profile.d/conda.sh" >> ~/.bashrc
</code></pre>

<p></p>

This makes sure that the upgraded environment settings are being applied.

<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke">source ~/.bashrc
</code></pre>

<p></p>

Now you are ready to use conda and load all environments we have built.

<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke">conda activate /home/j/jobersta/.conda/envs/genome.assembly
</code></pre>

<p></p>

Run the following code to generate the MultiQC Report. The input "FASTQ_QC/" is our path to the fastqc files and the output directory specified by "-o" is "MultiQC_Report". 

<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: #black"><code style = "background: black; color: whitesmoke">multiqc FASTQ_QC/ -o MultiQC_Report
</code></pre>

<p></p>

Lets use FileZilla to transfer this MultiQC_Report to our local server. To log in:
<p></p>
<strong>Host</strong>: sftp://sc.rc.usf.edu
<p></p>
<strong>Username</strong>: your USF net ID
<p></p>
<strong>Password</strong>: your USF net ID password
<p></p>

The right-hand side of FileZilla should show your files in the remote site(your student cluster) and the left-hand side shows all of your files in your local site. Make sure your local site is directed to the path you want your MultiQC report to be in then right click on the MultiQC_Report folder within the files under the remote site and click download. Launch the MultiQC_Report.html to see the summary statistics and the aggregated FastQC results. Below is a screenshot of what the report looks like. Sequence Counts is one of the graphs MultiQC displays but it also includes sequence quality histograms and per base content.

<p></p>

<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt2.png?raw=TRUE" width = 900>

<p></p>

Lets go back to running the rest of the scripts. 02_build_hisat_index.sh builds the index of our reference genome. <i>*it may be a good idea to write "module purge" before loading hisat2.</i> Remember these scripts are written in bash. In the following script, a path is being assigned to the variable, INPUT_REF_FASTA, highlighted in blue which is the fasta file of the genome we want to feed hisat2-build. The base name of the index files (*.ht2) is being assigned to the variable, "INDEX_FILES_BASE_NAME." To call these variables in bash, we use $. After building the index to hisat2, save and submit the job. The output files to go to our ./Genomics_Training/Work folder. 

<p></p>

<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt3.png?raw=TRUE" width = 900>

<p></p>

While in the ./Genomics_Training/Work directory, we can run the following code to check our hisat_index.out file. The "less" command displays the contents of a file. There should be no error messages.
<pre style = "background: black;border-style: solid;border-width: 2px; border-color: black"><code style = "background: black; color: whitesmoke">less hisat_index.out
</code></pre>

<p></p>

Now, we can allign our reads to this but first, make sure to go back to the ./Genomics_Training/My_Scripts folder before editing the script in vim. We are loading hisat2 again, to perform the alignment. In this script, you can see we are also loading an application called samtools. We will use this to organize the allignments from hisat2. The hisat2 allignment outputs sam(sequence alignment maps) files. We use use samtools to organze these files by position and convert them to bam files which are smaller. The script includes the variables we want but again, we have write the hisat command. The hisat2 command is made up of the following arguments:
<p></p>
<DD>--r : files wth unpaired reads</DD>
<p></p>
<DD>-p : a performance option defining the number of threads. ncreasing -p increases HISAT2’s memory footprint. E.g. when aligning to a human genome index, increasing -p from 1 to 8 increases the memory footprint by a few hundred megabytes. The number of threads we are using is defined by the NUMBER_OF_PROCESSOR variable. </DD>
<p></p>
<DD>--dta-cufflinks : Report alignments tailored specifically for Cufflinks. In addition to what HISAT2 does with the above option (–dta), With this option, HISAT2 looks for novel splice sites with three signals (GT/AG, GC/AG, AT/AC), but all user-provided splice sites are used irrespective of their signals. </DD>
<p></p>
<DD>-x : main argument; the basename of the index for the reference genome follows this. </DD>
<p></p>
<DD>--rna-strandness : specifies strand-specific information(default is unstranded). For single-end reads, use F or R. For paired-end read, use either FR or RF.</DD>
<p></p>
<DD>-1 : main argument; Comma-separated list of files containing mate 1s (filename usually includes _1), e.g. -1 flyA_1.fq,flyB_1.fq.</DD>
<p></p>
<DD>-2 : Comma-separated list of files containing mate 2s (filename usually includes _2), e.g. -2 flyA_2.fq,flyB_2.fq.</DD>
<p></p>
<DD>-S : File to write SAM alignments to</DD>
<p></p>

More information on the hisat2 commands can be found <a href="http://daehwankimlab.github.io/hisat2/manual/"> here</a>.

<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: black"><code style = "background: black; color: whitesmoke">hisat2 --r -p $NUMBER_OF_PROCESSOR --da-cufflinks -x $REF_GENOME --rna-strandess $RNA_STRANDNESS -1 $SAMPLE_R1 -2 $SAMPLE1_R2 -S $SAMPLE1_SAM_OUTFILE
</code></pre>

<p></p>

To break down the samtool command:
<p></p>
<DD>sort: sort alignments by leftmost coordinates</DD>
<p></p>
<DD>-@: number of threads</DD>
<p></p>
<DD>-o : where the sorted output is written to </DD>
<p></p>
More information on the samtools commands can be found <a href = "http://www.htslib.org/doc/samtools.html#COMMANDS">here</a>.
<p></p>

<pre style = "background: black;border-style: solid;border-width: 2px; border-color: black"><code style = "background: black; color: whitesmoke">samtools sort -@ $NUMBER_OF_PROESSORS -o $SAMPLE1_BAM_OUTFILE $SAMPLE1_SAM_OUTFILE
</code></pre>

<p></p>

These lines are repeated for all four samples. The first two are shown in this screenshot. After repeating the code for all four samples, save, and submit the job. You can can check for the bam files in the /Work output directory; there should be four .bam files.

<p></p>

<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt4.png?raw=TRUE" width = 900>

<p></p>
Next, we run cufflinks to assemble reads into transcripts. The cufflink command is made up of the following arguments:
<p></p>
<DD>-p : number of prosessors</DD>
<p></p>
<DD>-G : supplied reference annotation</DD>
<p></p>
<DD>-o : output files</DD>
<p></p>

More information on the cufflinks command can be found <a href = "http://cole-trapnell-lab.github.io/cufflinks/cufflinks/index.html">here</a>. We can use the command, "ls -lht", in the /Work directory to see the Cufflinks output. There should be four cufflinks folders for each sample. 

<p></p>
<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt5.png?raw=TRUE" width = 900>

<p></p>
Now, we get the normalized gene counts using cuffnorm. The cuffnorm command is similar to the cufflinks command:</DD>
<p></p>
<DD>–quiet: suppress messages other than serious warnings and errors.</DD>
<p></p>
<DD>–library-norm-method : specifies the library normalization method </DD>
<p></p>
<DD>-L : specifies a label for each sample</DD>
<p></p>
More information on the cuffnorm command can be found <a href = "http://cole-trapnell-lab.github.io/cufflinks/cuffnorm/index.html">here</a>.
The cuffnorm outputs in the /Work directory has its own folder inlcuding a set of files containing normalized expression levels for each gene, transcript, TSS group, and CDS group in the experiment. It does not perform differential expression analysis.
<p></p>
<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt6.png?raw=TRUE" width = 900>
<p></p>

Finally, we run featureCounts to get raw gene counts. The featureCounts command is made up of the following arguments:
<p></p>
<DD>-T : number of threads</DD>
<p></p>
<DD>-t : Specify the type of input sequencing data. Possible valuesinclude 0, denoting RNA-seq data, or 1, denoting genomic DNA-seq data.</DD>
<p></p>
<DD>-g : gene identifier</DD>
<p></p>
<DD>-a : name of annotation file</DD>
<p></p>
<DD>-o : name of output file</DD>
<p></p>
<DD>-p : If specified, fragments (or templates) will be counted instead
of reads.</DD>
<p></p>
More information on the featureCounts command can be found <a href = "https://bioconductor.org/packages/release/bioc/vignettes/Rsubread/inst/doc/SubreadUsersGuide.pdf">here</a>. It outputs numbers of reads assigned to features (or meta-features). It also outputs stat info for the overall summrization results, including number of successfully assigned reads and number of reads that failed to be assigned due to various reasons (these reasons are included in the stat info). We will be using the vehicle_drug_feature_counts.txt file tommorow!
<p></p>
<img src="https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/img/day2_jg_pt7.png?raw=TRUE" width = 900>
<p></p>
</div>


<h2 id="day-three">Day Three</h2>
<hr class = "custom">

<h3 id="presentation-slides">Presentation Slides</h3>


<table>
  <tr>
    <th>AGENDA</th>
    <th>INSTRUCTOR</th> 
    
  </tr>
  <tr>
    <td><a href= "" class="a.tablelink" download = "">Introduction to R</a></td>
    <td><span style = "color: white">Dr. Charley Wang</span></td>
  </tr>
  
  <tr>
  <td><a href = "" class="a.tablelink" download = "">RNA-Seq Allignment and Visualization</a></td>
<td><span style = "color: white">Dr. Charley Wang</span></td>  
  </tr>

  </table>

<h3 id="r-and-bioconductor-hands-on-practice">R and Bioconductor Hands-on Practice</h3>

Follow Charley's R Tutorials in order:
<p></p>
<ol>
  <li><a href = "https://github.com/usfomicshub/RNASeq_workshop_Sept2020/blob/master/tutorial/day2/R_code_Training_and_Reference/basic_R.html">The Basics : part 1</a> - Introducing basic R including all data types</li>
  <li><a href = "">The Basics : part 2</a> - How to load/save data and use control structures</li>
  <li><a href = "">Data Normalization</a> - How to normalize data and plot it</li>
  <li><a href = "">FPKM Correlation</a> - How to create heatmaps to check the correlation between each biological replicates</li>
  <li><a href ="">Differential Gene Expression</a> - How to perform differential gene expression analysis using DESeq2 and visualizing results using heatmaps<i>* use FileZilla to transfer our vehicle_drug_feature_counts.txt output file from day 2 to a path we can call</i></li>
  <li><a href = "">More DE Gene Visualizations</a> - How to create boxplots and histograms for significantly DE genes. 
</ol>




<h2 id ="day-four">Day Four</h2>
<hr class = "custom">

<h3 id="presentation-slides">Presentation Slides</h3>

<table>
  <tr>
    <th>AGENDA</th>
    <th>INSTRUCTOR</th> 
    
  </tr>
  <tr>
    <td><a href= "" class="a.tablelink" download = "">RNA-Seq: TUXEDO PIPELINE</a></td>
    <td><span style = "color: white">Dr. Jenna Oberstaller</span></td>
  </tr>
  
  <tr>
  <td><a href = "" class="a.tablelink" download = "">RNA-Seq: TUXEDO PIPELINE</a></td>
<td><span style = "color: white">Dr. Jenna Oberstaller</span></td>  
  </tr>

  </table>


<h3 id="rna-seq-tuxedo-pipeline">RNA-Seq: TUXEDO PIPELINE</h3>

<h3 id="ilab-review">iLab Review<h3> 




<!--


 <h2 id="resources">Resources</h2> 
  <hr class = "custom">
 <p>
 <p>Here are some resources mentioned in this workshop and some extra information that you might find helpful in your microbiome research. test</p>
  <p><details closed></p>
<p><summary>Journals for Statistical Considerations for Microbiome studies</summary></p>
<ul>
  </p><li><a href="https://msystems.asm.org/content/msys/4/4/e00325-19.full.pdf">A Cross-Sectional Study of Compositional and Functional Profiles of Gut Microbiota in Sardinian Centenarians</a></li></p>

<p><li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7040164/pdf/fnins-14-00127.pdf">Case-Control Study of the Effects of Gut Microbiota Composition on Neurotransmitter Metabolic Pathways in Children With Attention Deficit Hyperactivity Disorder</a></li></p>

<p><li><a href="https://www.mdpi.com/2076-2607/7/1/17/html">Socioeconomic Status and the Gut Microbiome: A TwinsUK Cohort Study</a></li></p>

<p><li><a href="https://www.nature.com/articles/s41366-019-0440-y?proof=true">Structured exercise alters the gut microbiota in humans with overweight and obesity—A randomized controlled trial</a></li></p>

<p><li><a href="https://www.nature.com/articles/s41467-017-01973-8">Meta-analysis of gut microbiome studies identifies disease-specific and shared responses</a></li></p>
</ul>
</details>
  <details closed>
  </p><summary>Microbiome R Packages</summary></p>
<ul>
<p><li><a href="https://www.bioconductor.org/packages/release/bioc/manuals/dada2/man/dada2.pdf">Dada2 Reference Manual</a></li></p>
<p><li><a href="https://github.com/picrust/picrust2/wiki/PICRUSt2-Tutorial-(v2.3.0-beta">PICRUSt2 Tutorial</a></li></p>
</ul>
  </details>
  </p>
  
     <h2 id="certificate-of-completion">Certificate Of Completion</h2> 
  <hr class = "custom">
  <p>Survey to receive your <span style = "color: #d28445">USF OmicsHub RNA-Seq Data Analysis Workshop Certificate of Completion</span> will be available at the end of the workshop</p>

  
  <details closed>  
  <p><summary>Please open this tab and take the survey to receive your <span style = "color: #d28445">USF Omicshub Microbiome Data Analysis Certificate</span>!</summary></p>
  <iframe src="https://docs.google.com/forms/d/e/1FAIpQLSdiBLf1qAT0FeqYW-At-vYziyQfakAeR7AvEippadrH5co8KQ/viewform?embedded=true" width="640" height="2990" frameborder="0" marginheight="0" marginwidth="0">Loading…</iframe>
  </details>
-->



