---
layout: default
---
<meta name="viewport" content="width=device-width, initial-scale=1">
<style>
img {
  display: block;
  margin-left: auto;
  margin-right: auto;
}
h1 {
  text-align: center;
  }
a:link {
  color: deepskyblue;
  background-color: transparent;
  text-decoration: none;
}
a:hover {
  color: deepskyblue ;
  background-color: transparent;
  text-decoration: underline;
}

a.tablelink:link {
  color: #DAF7A6;
  background-color: transparent;
  text-decoration: none;
  }
  a.tablelink:hover {
  color: #DAF7A6;
  background-color: transparent;
  text-decoration: underline;
}
  
.myDiv {
  padding: 10px;
  border-radius: 20px;
  background-color: #75b5aa;
  color: black;
}
.myDiv2 {
  padding: 10px;
  border-radius: 20px;
  background-color: lavender;
  color: black;
  }
 .myDiv3 {
   padding: 10px;
   border-radius: 20px;
   background-color: #aa759f;
   color: black;
  }
hr.custom {
  border-top: 1px dashed ;
  }
  table {
  width:100%;
}
table, th, td {
  
  border: 3px solid black;
  border-collapse: collapse;
  
}
th, td {
  padding: 15px;
  text-align: left;
}

th { 
  background-color: #303030;
  color: #f4bf75;
  }
  
tr {
 background-color: #006150
}  


.navbar {
  width: 100%;
  overflow: hidden;
  background-color: #005858;
  border: 4px solid #b5e853; 
  border-radius:10px
}

.navbar a {
  float: left;
  font-size: 16px;
  color: white;
  text-align: center;
  padding: 14px 16px;
  text-decoration: none;
}

.dropdown {
  float: left;
  overflow: hidden;
}

.dropdown .dropbtn {
  font-size: 16px;  
  border: none;
  outline: none;
  color: white;
  padding: 14px 16px;
  background-color: inherit;
  font-family: inherit;
  margin: 0;
}

.navbar a:hover, .dropdown:hover .dropbtn {
  background-color: #f4bf75;
}

.dropdown-content {
  display: none;
  position: absolute;
  background-color: #f9f9f9;
  min-width: 160px;
  box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
  z-index: 1;
}

.dropdown-content a {
  float: none;
  color: #005858;
  padding: 12px 16px;
  text-decoration: none;
  display: block;
  text-align: left;
}

.dropdown-content a:hover {
  background-color: #ddd;
}

.dropdown:hover .dropdown-content {
  display: block;
}

hr.custom {
  border-top: 1px dashed ;
  }
</style>


<a href="https://usfomicshub.github.io/">
<img src="https://github.com/usfomicshub/usfomicshub.github.io/blob/master/img/Hub_logo_2.0_small.PNG?raw=true" width="300" height="250" class="center">
</a>

<p></a></p>

<h1 align="center">Welcome to the USF Omics Hub git repository!</h1>

  <h3 align="center"><p style = "color: #f4bf75">We post our materials for our data-analysis workshops here, information for upcoming events, as well as the code we've developed for various other omics-related data analyses. </p></h3>

<div class="navbar">

  <div class="dropdown">
    <button class="dropbtn">USF Genomics
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics">USF Genomics Program</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/USF_Genomics_Core/index.html">USF Genomics Core: Facilities & Resources</a>
   </div>
  </div> 

  <div class="dropdown">
    <button class="dropbtn">Hub Service Inquires
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/request-services">Request Services</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Policies_and_Procedures/index.html">Service Policies & Procedures</a>
   </div>
  </div> 


  
  <div class="dropdown">
    <button class="dropbtn">Hub Workshops
      <i class="fa fa-caret-down"></i>
    </button>
    <div class="dropdown-content">
      <a href="https://health.usf.edu/publichealth/ghidr/genomics/genomics-training">Training Workshops Overview</a>
      <a href="https://usfomicshub.github.io/usfomicshub.github.io/Microbiome_Workshop_Materials/index.html">Microbiome Workshop (June 2020)</a>
   </div>
  </div> 


<a href="https://usfomicshub.github.io/usfomicshub.github.io/Cluster_Computing/index.html">Cluster Computing</a> 

<a href="https://usfomicshub.github.io/faq">FAQ</a> 


</div>


<p></p>
                                                                                                                                     
<h2 id="cluster-computing">Cluster Computing</h2>


<h3 id="table-of-contents">Table of Contents</h3>
<ol>
  <a href = "#usf-clusters"><li>USF Clusters<ul></a>
    <a href = "#creating-an-rra-cluster-account"><li>Creating an Account</li></a>

</ul>
</li>
<a href = "#hub-software-installs-on-usfs-rra-cluster"><li>Hub software installs on USFs RRA cluster<ul></a>
  <a href = "#available-software-and-tools"><li>Available Software and Tools</li></a>
  <a href = "#loading-and-using-software"><li>Loading and using Software</li></a>
</ul>
</li>
<a href="#more-about-usf-research-computing"><li>More about USF Research Computing<ul></a>
</ul>
</li>

</ol>

<h2 id="usf-clusters">USF Clusters</h2>
<hr class = "custom">
<p>USF Genomics researchers requiring high-performance computing have two primary cluster-computing options (both maintained by USF Research Computing):</p>

<p><span style = "color: #b5e853">&#11043;</span><span style = "color: #aa759f"><strong> Research-Restricted Access cluster (RRA)</strong></span>, to which USF Genomics Program members have priority access.</p>
<p><span style = "color: #b5e853">&#11043;</span><span style = "color: #aa759f"><strong> CIRCE</strong></span>, the primary cluster servicing all USF researchers.</p>

<br>
<h3 id="whats-the-difference-and-which-one-do-i-use" style = "color: #f4bf75">WHATS THE DIFFERENCE AND WHICH ONE DO I USE?</h3>
<p>Both clusters have excellent capabilities. As the name implies, <strong>RRA</strong> has stringent security measures in place to limit access and to protect potentially-sensitive human-subjects data. <em>Your research does not NEED to include HIPAA-protected data for you to use RRA.</em> Everybody requesting an RRA-account will, however, need to obtain HIPAA-training before their account can be created. Jobs run on RRA typically have faster turnaround-time as it is currently at lower utilization.</p>
<p>The <strong>CIRCE</strong> cluster, on the other hand, does not have as many restrictions on access and does not require additional certification for account-creation. Wait-times for job completions using CIRCE are likely to be longer due to much-higher utilization.</p>
<p>Please follow the instructions below to gain access to the appropriate cluster. Once the requirements have been satisfied, Research Computing is typically very quick to create your account.</p>
<br>
<h3 id="creating-an-rra-cluster-account" style = "color: #f4bf75">CREATING AN RRA CLUSTER ACCOUNT</h3>
<br>
<p>To request an RRA-Cluster account, email <a href="mailto:rc-help@usf.edu">rc-help@usf.edu</a> <em><strong>from your official USF email address</strong></em> with the following information:</p>
<ul>
<li>Supporting USF Faculty-member full name</li>
<li>Supporting USF Faculty-member email address</li>
</ul>
<p>Once contacted, RC administrators will send over all applicable SabaCloud/Docusign links/information, which will include instructions for completing your online HIPAA-certification. Note that <strong>HIPAA Training is <em>required</em> for access to the Genomics RRA cluster.</strong></p>
<p><em>If you or your supporting USF Faculty member are <strong>NOT</strong> listed on the <a href="https://health.usf.edu/publichealth/ghidr/genomics/researchers">USF Genomics Faculty page</a>, please email <a href="mailto:genomics@usf.edu">genomics@usf.edu</a> for clearance <strong>BEFORE</strong> requesting an account.</em></p>
<br>
<p><em><strong>All RRA accounts are automatically granted access to CIRCE. If requesting an RRA account, do not send in a request for a CIRCE account as well.</strong></em></p>
<br>
<br>
<h3 id="creating-a-circe-account" style = "color: #f4bf75">CREATING A CIRCE ACCOUNT</h3>
<br>
<p>To request a CIRCE account, email <a href="mailto:rc-help@usf.edu">rc-help@usf.edu</a> <em><strong>from your official USF email address</strong></em> with the following information:</p>
<ul>
<li>Supporting USF Faculty-member full name</li>
<li>Supporting USF Faculty-member email address</li>
</ul>
<p><em><strong>NOTE: Only send a request for a CIRCE account if you do not currently have (or are not actively requesting) an RRA account.</strong></em></p>
<br>
<br>
<br>
<p>You can refer to <a href="https://wiki.rc.usf.edu/index.php/Main_Page">Research Computing's documentation-wiki</a> for further information on HPC-basics, CIRCE hardware and other useful topics. Refer to <a href="https://github.com/usfomicshub/usfomicshub.github.io/blob/master/wiki/markdowns/facilities_and_resources.md">our Genomics Facilities and Resources</a> for information on RRA hardware and specifications.</p>


<h2 id="hub-software-installs-on-usfs-rra-cluster">Hub software installs on USF's RRA cluster</h2>
<hr class = "custom">
<p>USF Research Computing maintains the High-Performance Computing (HPC) hardware and resources available to all USF Researchers.  They have gone above and beyond to provide installations of USF Genomics software-requests on both CIRCE and RRA clusters. Their list of available software and any documentation can be found <a href=https://wiki.rc.usf.edu/index.php/Applications>here</a>. A more updated list of pending and recently-completed installs can be found <a href=https://wiki.rc.usf.edu/index.php/Apps_Queue>here</a>, though most of these have not yet been documented due to the bulk of requests.</p>
<p>To relieve some of the Genomics-software installation- and documentation-burden from RC, the Hub (with guidance from RC) has established an independently-maintained repository on RRA of mostly omics-related software available for all RRA account-holders.</p>
<h3 id="available-software-and-tools" style = "color: #f4bf75">AVAILABLE SOFWARE AND TOOLS</h3>
<h4 id="anaconda3-environments">Anaconda3 environments</h4>
<ul>
<li>humann2
<ul>
<li>(version 2.8.1)</li>
<li><a href="http://huttenhower.sph.harvard.edu/humann">HUMAnN2: The HMP Unified Metabolic Analysis Network 2 documentation</a></li>
</ul>
</li>
<li>fastq_dump</li>
<li>kera2</li>
<li>lumpy
<ul>
<li>(version 0.2.13)</li>
</ul>
</li>
<li>picrust2
<ul>
<li>(version 2.3.0_b)</li>
<li><a href="https://github.com/picrust/picrust2/wiki">PICRUSt2 documentation</a></li>
</ul>
</li>
<li>Rdeseq
<ul>
<li>R version 3.6.2 (2019-12-12)</li>
</ul>
</li>
<li>selene</li>
<li>qiime2-2019.7</li>
</ul>
<h4 id="anaconda2-environments">Anaconda2 environments</h4>
<ul>
<li>epiGBS_shared</li>
<li>kera2</li>
</ul>
<h4 id="homebrew-packages">Homebrew packages</h4>
<ul>
<li>homebrew</li>
</ul>
<h4 id="other-load-via-hub-supplied-modules">Other (load via Hub-supplied modules)</h4>
<ul>
<li>SeekDeep</li>
</ul>
<h3 id="loading-and-using-software" style = "color: #f4bf75">LOADING AND USING SOFTWARE</h3>
<p>Omics data-analysis workflows often require chaining together several different programs, each with their own dependencies, quirks, and incompatibilities. For this reason, many programs commonly used together can be loaded all together in a virtual &quot;environment&quot; specifically designed to &quot;just work&quot; without the incompatibility-headaches. We use Anaconda to install, create and manage these virtual environments for you.</p>
<h3 id="loading-hub-anaconda-environments-on-rra">Loading Hub Anaconda environments on RRA</h3>
<p>Best practice when loading any environment or module is to start by purging your current environment to minimize the chance of loading incompatible programs.</p>
<pre><code>  module purge
</code></pre>
<h4 id="the-first-time-you-use-the-hub-anaconda-install-do-these-three-steps">The FIRST time you use the Hub Anaconda-install, do these three steps:</h4>
<h5 id="add-conda-to-your-path-for-this-login-session-and-permanently-so-you-never-have-to-do-this-again">1. add conda to your path for this login-session AND permanently so you never have to do this again.**</h5>
<p><em>Your file-path is searched for programs etc. by the order in which directories appear in your path-variable. Pre-pending to your path ensures that the version of anaconda you load is the one you specify.</em></p>
<ul>
<li>
<p>if your environment is provided via anaconda3 (the appropriate choice for most users):</p>
<pre><code>export PATH=/shares/omicshub/apps/anaconda3/bin:$PATH &gt;&gt; ~/.bashrc
</code></pre></li>
<li>
<p>if your environment is provided via anaconda2:</p>
<pre><code>export PATH=/shares/omicshub/anaconda2/bin:$PATH &gt;&gt; ~/.bashrc
</code></pre>
<p>**<em>Only use this option of permanently setting your anaconda-version if you're sure that's the one you want to use. If you need to switch back and forth between environments available via anaconda2 and anaconda3, it is better to set your path ONLY for your current login-session (meaning without appending the anaconda path to your environment-settings in ~/.bashrc). This example will set your path only for the current session:</em></p>
<pre><code> export PATH=/shares/omicshub/anaconda2/bin:$PATH
</code></pre></li>
</ul>
<h5 id="initialize-conda">2. initialize conda</h5>
<p>This command tells conda to automatically add a block of code to your default environment-settings--your ~/.bashrc file--so you never have to do this again.</p>
<pre><code>conda init
</code></pre>
<h5 id="tell-it-to-use-the-environment-settings-you-just-upgraded">3. tell it to use the environment-settings you just upgraded</h5>
<pre><code>  source ~/.bashrc
</code></pre>
<p><em>alternatively you can logout and login to RRA again and your new settings will take effect automatically.</em></p>
<p>That's it! Now you are ready to use conda and all environments we have built.</p>
<h3 id="using-hub-anaconda-environments-on-rra">Using Hub Anaconda environments on RRA</h3>
<p>List the environments available to load:</p>
<pre><code>  conda env list
</code></pre>
<p>Load the environment (using qiime2 as an example):</p>
<p><em>loading an environment adds all the programs necessary to run a given analysis-pipeline directly to your path.</em></p>
<pre><code>  conda activate qiime2-2019.7
</code></pre>
<p>Each environment will have many programs/packages/tools installed. To see packages and their versions installed in the loaded environment:</p>
<pre><code>  conda list
</code></pre>
<p>Sometimes you will want to check the available executable commands in an environment. All executables are stored in the same place for each anaconda3 environment so you can easily check. Using qiime2-2019.7 as an example:</p>
<pre><code>  ls -l /shares/omicshub/apps/anaconda3/envs/qiime2-2019.7/bin
</code></pre>
<p><em>This command will list all the executables installed in the qiime2-2019.7 environment. Typically these file-names are the commands you can use from programs loaded in the given environment.</em></p>
<p>Unloading the environment:</p>
<p><em>When you are finished using the environment, you should unload it to avoid introducing compatibility-issues with software-versions, etc.</em></p>
<pre><code>  conda deactivate
</code></pre>

<h2 id="more-about-usf-research-computing">More about USF Research Computing</h2>
<hr class = "custom">
<p>You can find more information about USF's clusters as well as Linux tutorials and high performance computing basics on USF's Research Computing main wiki page <a href=https://wiki.rc.usf.edu/index.php/Main_Page>here</a>.